pkg/utils/utils.go

```diff
@@ -23,3 +23,16 @@ func GetMapValue(key string, dataMap map[string][]byte) (string, bool) {
 	}
 	return "", false
 }
+
+// Add key-value to map
+func AddMapValue(key string, value string, dataMap map[string]string, overwrite bool) map[string]string {
+	if _, ok := dataMap[key]; ok && !overwrite {
+		return dataMap
+	} else if dataMap != nil {
+		dataMap[key] = value
+		return dataMap
+	}
+	dataMap = map[string]string{}
+	dataMap[key] = value
+	return dataMap
```

Comment:
@S-mishina 

Oh... it's good to handle the case where dataMap is nil 👍 

It looks good in general but let's make it more straightforward and slimmer (i mean L32 and L36 ).  
How about this?

```suggestion
	if _, ok := dataMap[key]; ok && !overwrite {
		return dataMap
	}
	if dataMap == nil {
		dataMap = map[string]string{}
	}
	dataMap[key] = value
	return dataMap

```

Reply:
@yokawasa 
Thanks for the review!
That's what I wrote at first too!
I will adopt that one.